src/microbe_stage/player_hover_info.cpp

```diff
@@ -65,19 +65,31 @@ void
             "noCompounds", new Leviathan::BoolBlock(true)));
 
     } else {
-        for(const auto& tuple : compounds) {
 
-            std::stringstream compoundInfo;
-            compoundInfo << std::fixed << std::setprecision(2)
-                         << SimulationParameters::compoundRegistry
-                                .getTypeData(std::get<0>(tuple))
-                                .displayName
-                         << ": " << std::get<1>(tuple);
+        Json::Value compoundsJson;
+
+        for(const auto& tuple : compounds) {
 
-            vars->Add(std::make_shared<NamedVariableList>(
-                "compound" + std::to_string(std::get<0>(tuple)),
-                new Leviathan::StringBlock(compoundInfo.str())));
+            Json::Value compound;
+            Json::Value quantity;
+            Json::Value name;
+            compound["name"] = SimulationParameters::compoundRegistry
+                                   .getTypeData(std::get<0>(tuple))
+                                   .displayName;
+            compound["quantity"] = std::to_string(std::get<1>(tuple));
```

Comment:
There's unused variables here, also the float should be passed as is, so that the code doing the displaying is allowed to do decisions about how many decimals to print etc.
```
        for(const auto& tuple : compounds) {

            Json::Value compound;
            compound["name"] = SimulationParameters::compoundRegistry
                                   .getTypeData(std::get<0>(tuple))
                                   .displayName;
            compound["quantity"] = std::get<1>(tuple);
            compoundsJson.append(compound);
        }
```